turbo/engineapi/engine_server.go

```diff
@@ -944,3 +958,18 @@ func (e *EngineServer) HandlesForkChoice(
 	}
 	return payloadStatus, nil
 }
+
+func waitForStuff(waitCondnF func() (bool, error)) (bool, error) {
```

Comment:
there is a library for this type of things that we have available in our repo and should use - https://pkg.go.dev/github.com/cenkalti/backoff/v4#RetryWithData

Reply:
example usage, https://github.com/erigontech/erigon/blob/main/polygon/p2p/fetcher_base.go#L480